### ISSUE TITLE
[FEAT] Add `valid_loss` option to models.

### ISSUE THREAD

**Issue by kdgutier** [Nixtla/neuralforecast#407]
To improve hyperparameter optimization one may want to maximize correlation between the validation loss and the final test loss.
To do so we can add a `valid_loss` argument to overwrite the current default `valid_loss=loss`.
